t/e2e/services/debug_api_test.go

```diff
@@ -0,0 +1,152 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/status-im/status-go/node"
+	"github.com/status-im/status-go/params"
+	"github.com/stretchr/testify/suite"
+
+	. "github.com/status-im/status-go/t/utils"
+)
+
+func TestDebugAPISuite(t *testing.T) {
+	s := new(DebugAPISuite)
+	s.upstream = false
+	suite.Run(t, s)
+}
+
+func TestDebugAPISuiteUpstream(t *testing.T) {
+	s := new(DebugAPISuite)
+	s.upstream = true
+	suite.Run(t, s)
+}
+
+type DebugAPISuite struct {
+	BaseJSONRPCSuite
+	upstream bool
+}
+
+func (s *DebugAPISuite) TestAccessibleDebugAPIsUnexported() {
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, false)
+	s.NoError(err)
+	// Debug APIs should be unavailable
+	s.AssertAPIMethodUnexported("debug_postSync")
+	err = s.Backend.StopNode()
+	s.NoError(err)
+
+	err = s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+	// Debug APIs should be available
+	s.AssertAPIMethodExported("debug_postSync")
+}
+
+func (s *DebugAPISuite) TestDebugPostSyncSuccess() {
+	// Test upstream if that's not StatusChain
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+
+	s.addPeerToCurrentNode()
+	symID := s.generateSymKey()
+	result := s.sendPostConfirmMessage(symID)
+
+	var r struct {
+		Error struct {
+			Message string `json:"message"`
+		} `json:"error"`
+		Result hexutil.Bytes `json:"result"`
+	}
+	s.NoError(json.Unmarshal([]byte(result), &r))
+	s.Empty(r.Error.Message)
+	s.NotEmpty(r.Result)
+}
+
+// generateSymKey generates and stores a symetric key.
+func (s *DebugAPISuite) generateSymKey() string {
+	w, err := s.Backend.StatusNode().WhisperService()
+	s.Require().NoError(err)
+	symID, err := w.GenerateSymKey()
```

Comment:
Doesn't `GenerateSymKey(password)` store the sym key in Whisper service already? `shh_addSymKey` seems unnecessary by looking at the implementation.